matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/timeline/TimelineEventDecryptor.kt

```diff
@@ -65,26 +66,30 @@ internal class TimelineEventDecryptor(
         cryptoService.removeSessionListener(newSessionListener)
         executor?.shutdownNow()
         executor = null
-        unknownSessionsFailure.clear()
-        existingRequests.clear()
+        synchronized(unknownSessionsFailure) {
+            unknownSessionsFailure.clear()
+        }
+        synchronized(existingRequests) {
+            existingRequests.clear()
+        }
     }
 
     fun requestDecryption(eventId: String) {
+        synchronized(unknownSessionsFailure) {
+            for (eventIds in unknownSessionsFailure.values) {
+                if (eventId in eventIds) {
+                    Timber.d("Skip Decryption request for event $eventId, unknown session")
+                    return
+                }
+            }
+        }
         synchronized(existingRequests) {
             if (eventId in existingRequests) {
```

Comment:
Not strictly part of the PR.... but this could be `if (!existingRequests.add(eventId))`.

Reply:
ok